### PR TITLE
Standardizes shunting to arrivals shuttle

### LINKED ISSUE #27957: people are getting put INSIDE chairs on the arrivals shuttle
WHAT THE FUCK

### PR DESCRIPTION
**Pull request by QualityVan**
:cl: QualityVan
fix: Stopped people getting stuck inside chairs on the arrivals shuttle
/:cl:

So GLOB.latejoin contains a list of chairs. A lot of stuff that shunts there, mostly secondary in case trying to shunt somewhere else fails, treated it as a list of turfs. This fixes that and prevents it from breaking if someone destroys all the chairs on the arrivals shuttle. Adds a get_latejoin_turf() proc that returns a suitable turf on the arrivals shuttle.

Fixes #27957

### CODE REVIEW

File: code/__HELPERS/game.dm [hunk 1]
```diff
@@ -506,14 +506,31 @@
...
 
 	return new_character
 
+/proc/get_latejoin_turf()
+	var/turf/D
+	if(GLOB.latejoin.len)
+		D = get_turf(pick(GLOB.latejoin))
+	if(!D)
+		for(var/turf/T in get_area_turfs(/area/shuttle/arrival))
+			if(!T.density)
```
Comment (AnturK) [inline code/__HELPERS/game.dm]: There's a helper for this check, `is_turf_blocked` is the name iirc

File: code/datums/mind.dm [hunk 2]
```diff
@@ -1448,7 +1448,7 @@
 		special_role = "Wizard"
 		assigned_role = "Wizard"
 		if(!GLOB.wizardstart.len)
-			current.loc = pick(GLOB.latejoin)
+			current.loc = get_latejoin_turf()
```
Comment (AnturK) [inline code/datums/mind.dm]: Make this forceMove while you're at it.

File: code/modules/admin/admin.dm [hunk 3]
```diff
@@ -595,7 +595,7 @@
 	set category = "Admin"
 	set name = "Unprison"
 	if (M.z == ZLEVEL_CENTCOM)
-		M.loc = pick(GLOB.latejoin)
+		M.loc = get_latejoin_turf()
```
Comment (AnturK) [inline code/modules/admin/admin.dm]: Same here.

File: code/modules/mob/dead/new_player/new_player.dm [hunk 4]
```diff
@@ -322,22 +322,7 @@
...
-				for(var/obj/O in T)
-					if(O.density)
-						clear = 0
-						break
-				if(clear)
-					D = T
-					continue
-
-	character.loc = D
+	character.loc = get_latejoin_turf()
```
Comment (AnturK) [inline code/modules/mob/dead/new_player/new_player.dm]: And here.

**Review by Cyberboss — CHANGES_REQUESTED** [review]
There are WAAYYYY more instances than this. See #26647

**Comment by Cyberboss** [comment]
Ugh, I'll stop being lazy and finish it...

**Comment by GunHog** [comment]
**GLOB.**

**Comment by AnturK** [comment]
First PR to remove conflicts gets merged @Cyberboss @QualityVan

**Pull request closed without merging**